### ISSUE TITLE
Errors: RuntimeError: protection exception (16777763) / 'gbk' codec can't decode byte / RuntimeError: invalid v8 code

### ISSUE THREAD

**Issue by JiaPai12138** [dashingsoft/pyarmor#999]
Pyarmor 8.0.9 (pro), 00xxxx, non-profits
Command:
```
pyarmor gen --assert-call test.py
```
test.py:
```
from win32con import VK_LEFT, VK_RIGHT, VK_END
from ctypes import windll
from win32api import GetAsyncKeyState
from sys import exit, executable
from time import sleep
import os


if __name__ == '__main__':
    if not windll.shell32.IsUserAnAdmin():
        windll.shell32.ShellExecuteW(None, 'runas', executable, __file__, None, 1)
        exit(0)

    try:
        windll.shcore.SetProcessDpiAwareness(1)
    except:
        try:
            windll.user32.SetProcessDPIAware()
        except:
            pass

    os.system('')
    print('Raise error'.encode('utf-8'))
    language_using = windll.kernel32.GetUserDefaultUILanguage()

    if language_using in [int('0xc04', 16), int('0x804', 16), int('0x404', 16)]:
        instruction_right = '[+] 按方向右键 -> 向右移动100单位'
        instruction_left = '[+] 按方向左键 <- 向左移动10单位'
        instruction_use = '[+] 请先按右方向键转一圈, 再按左方向键微调'
        instruction_moved = '目前总共移动'
        instruction_end = '[+] 移动完毕'
    else:
        instruction_right = '[+] Press the right arrow key -> to move right 100 pixels'
        instruction_left = '[+] Press the left arrow key <- to move left 10 pixels'
        instruction_use = '[+] Please move around by turning right, then adjust by moving left'
        instruction_moved = 'Totally moved'
        instruction_end = '[+] Completed'

    total_move = 0
    print(instruction_right)
    print(instruction_left)
    print(instruction_use)
    print('==============================')

    while True:
        windll.winmm.timeBeginPeriod(1)
        windll.kernel32.Sleep(200)
        windll.winmm.timeEndPeriod(1)

        if GetAsyncKeyState(VK_RIGHT)<0:
            total_move += 100
            print('[+] '+instruction_moved, total_move, end='\r')
        elif GetAsyncKeyState(VK_LEFT)<0:
            total_move -= 10
            print('[-] '+instruction_moved, total_move, end='\r')
        elif GetAsyncKeyState(VK_END)<0:
            print(instruction_end, '                        ')
            print('==============================')
            text_colors = ['42', '43', '44', '46', '47']
            for i in range(10):
                text_color = text_colors[i%len(text_colors)]
                print(instruction_moved, '\033[2;30;{}m '.format(text_color)+str(total_move)+' \033[0;0m', end='\r')
                sleep(0.3)
                print('                        ', end='\r')
                sleep(0.2)
            break

    exit(0)

```
Output:
```
$ test
b'Raise error'
[+] Press the right arrow key -> to move right 100 pixels
[+] Press the left arrow key <- to move left 10 pixels
[+] Please move around by turning right, then adjust by moving left
==============================
Traceback (most recent call last):
  File "<E:\test.py>", line 3, in <module>
  File "<test>", line 50, in
RuntimeError: protection exception (16777763)
```

Also when i was trying, i have to modify pyarmor\cli\resouce.py:
![image](https://user-images.githubusercontent.com/72233260/227249299-c5abe356-4238-40b7-8828-b58b63517505.png)
Or it will show me something like gbk encoding errors like:
```
'gbk' codec can't decode byte 0xaf in position 5370: illegal multibyte sequence
```

I could also get this error while encrypting some other scripts:
```
3752 MainProcess 2023-03-23 11:17:37,382
Traceback (most recent call last):
  File "C:\Python310\lib\site-packages\pyarmor\cli\__main__.py", line 559, in main
    main_entry(sys.argv[1:])
  File "C:\Python310\lib\site-packages\pyarmor\cli\__main__.py", line 547, in main_entry
    args.func(ctx, args)
  File "C:\Python310\lib\site-packages\pyarmor\cli\__main__.py", line 144, in cmd_gen
    builder.process(options, pack=args.pack)
  File "C:\Python310\lib\site-packages\pyarmor\cli\generate.py", line 153, in process
    self._obfuscate_scripts()
  File "C:\Python310\lib\site-packages\pyarmor\cli\generate.py", line 118, in _obfuscate_scripts
    code = Pytransform3.generate_obfuscated_script(self.ctx, r)
  File "C:\Python310\lib\site-packages\pyarmor\cli\core\__init__.py", line 48, in generate_obfuscated_script
    return m.generate_obfuscated_script(ctx, res)
  File "<maker>", line 711, in generate_obfuscated_script
  File "<maker>", line 515, in process
  File "<maker>", line 515, in <listcomp>
  File "<maker>", line 1731, in handle
  File "<maker>", line 1728, in visit
  File "<maker>", line 1725, in visit
  File "<maker>", line 1691, in patch_co_object
  File "<maker>", line 1487, in _patch_co_object
RuntimeError: invalid v8 code
```
example.py that could cause this error:
```
from ctypes import windll, c_bool, c_void_p, c_long, c_ulong, c_ushort, Structure, Union, c_int, POINTER, POINTER, sizeof
from sys import exit, executable


LONG = c_long
WORD = c_ushort
DWORD = c_ulong
ULONG_PTR = POINTER(DWORD)


class MOUSEINPUT(Structure):
    _fields_ = [('dx', LONG),
                ('dy', LONG),
                ('mouseData', DWORD),
                ('dwFlags', DWORD),
                ('time', DWORD),
                ('dwExtraInfo', ULONG_PTR)]


class KEYBDINPUT(Structure):
    _fields_ = [('wVk', WORD),
                ('wScan', WORD),
                ('dwFlags', DWORD),
                ('time', DWORD),
                ('dwExtraInfo', ULONG_PTR)]


class HARDWAREINPUT(Structure):
    _fields_ = [('uMsg', DWORD),
                ('wParamL', WORD),
                ('wParamH', WORD)]


class _INPUTunion(Union):
    _fields_ = [('mi', MOUSEINPUT),
                ('ki', KEYBDINPUT),
                ('hi', HARDWAREINPUT)]

class INPUT(Structure):
    _fields_ = [('type', DWORD),
                ('union', _INPUTunion)]


def SendInput(*inputs):
    nInputs = len(inputs)
    LPINPUT = INPUT * nInputs
    pInputs = LPINPUT(*inputs)
    cbSize = c_int(sizeof(INPUT))
    return windll.user32.SendInput(nInputs, pInputs, cbSize)


def Input(structure):
    if isinstance(structure, MOUSEINPUT):
        return INPUT(0, _INPUTunion(mi=structure))
    if isinstance(structure, KEYBDINPUT):
        return INPUT(1, _INPUTunion(ki=structure))
    if isinstance(structure, HARDWAREINPUT):
        return INPUT(2, _INPUTunion(hi=structure))
    raise TypeError(b'Cannot create INPUT structure!'.decode())


def MouseInput(flags, x, y, data):
    return MOUSEINPUT(x, y, data, flags, 0, None)


def KeybdInput(code, flags):
    return KEYBDINPUT(code, code, flags, 0, None)


def HardwareInput(message, parameter):
    return HARDWAREINPUT(message & 0xFFFFFFFF,
                        parameter & 0xFFFF,
                        parameter >> 16 & 0xFFFF)


def Mouse(flags, x=0, y=0, data=0):
    return Input(MouseInput(flags, x, y, data))


def Keyboard(code, flags=0):
    return Input(KeybdInput(code, flags))


def Hardware(message, parameter=0):
    return Input(HardwareInput(message, parameter))


if __name__ == '__main__':
    if not windll.shell32.IsUserAnAdmin():
        windll.shell32.ShellExecuteW(None, 'runas', executable, __file__, None, 1)
        exit(0)

    pass
```

**Comment by jondy** [comment]
Got it, thanks. I'll check it.

**Comment by jondy** [comment]
Which Python version is used for `RuntimeError: invalid v8 code`? I can't reproduce it with Python 3.7

**Comment by jondy** [comment]
Which Python version is used for `RuntimeError: invalid v8 code`? I can't reproduce it with Python 3.7

**Comment by JiaPai12138** [comment]
> Which Python version is used for `RuntimeError: invalid v8 code`? I can't reproduce it with Python 3.7

Python 3.10

**Comment by jondy** [comment]
Reproduced, thanks.

**Comment by jondy** [comment]
Fixed in 8.0.10.

Note that `--assert-call` may not work in complex scripts.  Sometimes it need to do some extra configuration by command `cfg`, I'll document the usage later.

**Comment by JiaPai12138** [comment]
Many thanks!
